src/OrbitQt/MoveFilesProcess.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitQt/MoveFilesProcess.h"
+
+#include "OrbitBase/File.h"
+#include "OrbitBase/Logging.h"
+#include "Path.h"
+
+namespace orbit_qt {
+
+MoveFilesProcess::MoveFilesProcess() : QObject(nullptr) {
+  background_thread_.start();
+  moveToThread(&background_thread_);
+}
+
+MoveFilesProcess::~MoveFilesProcess() noexcept {
+  background_thread_.quit();
+  background_thread_.wait();
+}
+
+void MoveFilesProcess::Run() {
+  QMetaObject::invokeMethod(this, [this]() { RunImpl(); });
+}
+
+void MoveFilesProcess::ReportError(const std::string& error_message) {
+  ERROR("%s", error_message);
+  emit generalError(error_message);
+}
+
+void MoveFilesProcess::TryMoveFilesAndRemoveDirIfNeeded(const std::filesystem::path& src_dir,
+                                                        const std::filesystem::path& dest_dir) {
+  auto file_exists_or_error = orbit_base::FileExists(src_dir);
+  bool file_exists = true;
+  if (file_exists_or_error.has_error()) {
+    ReportError(absl::StrFormat("Unable to stat \"%s\": %s", src_dir.string(),
+                                file_exists_or_error.error().message()));
+  } else {
+    file_exists = file_exists_or_error.value();
+  }
+
+  if (!file_exists) {
+    return;
+  }
```

Comment:
what about `else if (!file_exists_or_error.value()) { return; }`

This would avoid the extra `bool file_exists`

Reply:
Or is there a return missing in the `ReportError` case above?